src/notecard_platform.c

```diff
@@ -22,6 +23,46 @@ static bool g_i2c_initialized = false;
 static bool g_uart_initialized = false;
 static i2c_master_bus_handle_t g_i2c_bus_handle = NULL;
 
+// FreeRTOS mutex handles for thread-safe access
+#ifdef CONFIG_NOTECARD_I2C_MUTEX
+static SemaphoreHandle_t g_i2c_mutex = NULL;
+#endif
+static SemaphoreHandle_t g_notecard_mutex = NULL;
+
+//=============================================================================
+// Mutex Platform Implementation
+//=============================================================================
+
+#ifdef CONFIG_NOTECARD_I2C_MUTEX
+static void notecard_platform_lock_i2c(void)
+{
+    if (g_i2c_mutex != NULL) {
+        xSemaphoreTake(g_i2c_mutex, portMAX_DELAY);
```

Comment:
Poor man's error handling...

```suggestion
        for (;xSemaphoreTake(g_i2c_mutex, portMAX_DELAY) != pdTRUE;);
```